### PR TITLE
modified belief parameter type

### PR DESCRIPTION
**Pull request by LorenzoBonanni**
Modified belief parameter type to make it as general as possible.
I've noticed that when using Despot or POMCP both save the belief in a custom structure that extends AbstractParticleBelief and also Particle Collection extends it so I thought I should change the method signature to be as general as possible.

### CODE REVIEW

File: src/actions.jl
Comment (zsunberg) [inline src/actions.jl]: Oh, I should have caught this earlier! I think we should actually make it as general as possible with this
```suggestion
function POMDPs.actions(pomdp::RockSamplePOMDP, b)
```
(Also, no need for `K`)

**Review by zsunberg — CHANGES_REQUESTED** [review]
Since the concept of a belief is extremely flexible in POMDPs.jl, I think we should accept anything that is not a state - see my suggestion below

File: src/actions.jl
Comment (LorenzoBonanni) [reply src/actions.jl]: Ok that makes sense

**Comment by LorenzoBonanni** [comment]
@zsunberg after this merge you should create a new release

**Comment by zsunberg** [comment]
Thanks - will merge after tests pass

**Pull request merged**
